docs/development/verilog_modules.md

```diff
@@ -0,0 +1,122 @@
+
+As of now, there is only one submodule and its ports are connected directly to
+external pins, even those that have special functions and shouldn't be used as
+generic IO. This was done to test if the design can be successfully placed and
+routed, and to get rough estimates of FPGA utilization, but it doesn't produce
+usable board yet.
+
+After other modules are added and interconnected, this section will show mapping
+of interface signals (LPC or SPI) to board's IO pins.
+
+Current FPGA utilization:
+
+```text
+Device Utilization: 0.14 (target 1.00)
+	Physical Tile TL-LOGIC:
+	Block Utilization: 0.16 Logical Block: PB-LOGIC
+	Physical Tile TL-RAM:
+	Block Utilization: 0.00 Logical Block: PB-RAM
+	Physical Tile TL-MULT:
+	Block Utilization: 0.00 Logical Block: PB-MULT
+	Physical Tile TL-BIDIR:
+	Block Utilization: 0.94 Logical Block: PB-BIDIR
+	Physical Tile TL-CLOCK:
+	Block Utilization: 0.00 Logical Block: PB-CLOCK
+	Physical Tile TL-SDIOMUX:
+	Block Utilization: 0.79 Logical Block: PB-SDIOMUX
+	Physical Tile TL-GMUX:
+	Block Utilization: 0.20 Logical Block: PB-GMUX
+	Physical Tile TL-ASSP:
+	Block Utilization: 1.00 Logical Block: PB-ASSP
+	Physical Tile TL-SYN_VCC:
+	Block Utilization: 1.00 Logical Block: PB-SYN_VCC
+	Physical Tile TL-SYN_GND:
+	Block Utilization: 1.00 Logical Block: PB-SYN_GND
+```
+
+## LPC module
+
+This module is responsible for managing LPC communication. It responds only to
+TPM cycles, other cycle types are ignored. SERIRQ (both continuous and quiet
+mode), cycle aborts and LPC resets are implemented.
+
+```text
+   LPC interface:                                   Data provider interface:
+                           +-------------+
+            clk_i  ------->|             |>=======  lpc_addr_o [15:0]
+           nrst_i  ------->|             |<>======  lpc_data_io [7:0]
+         lframe_i  ------->|             |>-------  lpc_data_wr
+    lad_bus [3:0]  ======<>|     LPC     |<-------  lpc_wr_done
+           serirq  ------<>|   module    |>-------  lpc_data_req
+                           |             |<-------  lpc_data_rd
+                           |             |<=======  irq_num [3:0]
+                           |             |<-------  interrupt
+                           +-------------+
+```
+
+Ports for LPC interface (refer to [LPC specification](https://www.intel.com/content/dam/www/program/design/us/en/documents/low-pin-count-interface-specification.pdf)
+for details):
+
+- `input clk_i`: LPC clock.
+- `input nrst_i`: LPC reset (active low).
+- `input lframe_i`: LPC frame input (active low).
+- `inout lad_bus`: LPC data bus, slow pull-up on host side.
+- `inout serirq`: LPC SERIRQ signal, slow pull-up on host side. More information
+  about SERIRQ can be found [here](https://web.archive.org/web/20150502224201/http://hackipedia.org/Platform/x86/PCI/Serialized%20IRQ%20Support%20for%20PCI%20Systems.doc.pdf)
+  or in PCI specifications (not publicly available).
```

Comment:
which PCI specification? I couldn;t find SERIRQ mentions in PCI local bus specification nor PCI to PCI bridge architecture specification

Reply:
I can't find it anymore. Removed that part.